packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -261,20 +258,28 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
     _focusNode = FocusNode();
   }
 
+  void expandParents(T parent) {
+    if (parent?.parent?.index != -1) {
+      expandParents(parent?.parent);
+    }
+
+    if (parent != null && !parent.isExpanded) {
+      _toggleNode(parent);
+    }
+  }
+
   @override
   void didUpdateWidget(TreeTable oldWidget) {
     super.didUpdateWidget(oldWidget);
 
     cancel();
 
     addAutoDisposeListener(selectionNotifier, () {
-      selectionNotifier.value.displayRow = _selectionRowNumber();
-
       setState(() {
         final node = selectionNotifier.value.node;
         final parent = node.parent;
-        if (!parent.isExpanded) {
-          _toggleNode(parent);
+        if (parent != null) {
```

Comment:
all this code could just be
expandParents(node.parent);
if the null check was added within the method.

Reply:
Right, done.